cgrok/grokre.c

```diff
@@ -383,6 +383,17 @@ static void grok_capture_add_predicate(grok_t *grok, int capture_id,
   grok_capture_add(grok, gct, renamed_only);
 }
 
+/* Parse the group name, up to the max length, and try to parse it as a hex number.
+   Return 0 if the string is not a valid number, or 1 otherwise.*/
+static int grok_parse_capture_id(char* groupName, int *id) {
+	char *endOfNum;
+	if (groupName[0] != '0' || groupName[1] != 'x') {
+		return 0;
+	}
+	*id = strtol(groupName, &endOfNum, 16);
+	return (endOfNum-groupName) == 6;
```

Comment:
Do you mind checking `errno` here? I can see you're checking the length of the string to ensure it is the length of `CAPTURE_LEN`, but we don't actually know if `strtol` succeeded or not.